pkg/status/bird_status.go

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"time"
+
+	apiv3 "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
+
+	"github.com/olekukonko/tablewriter"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// birdStatus is a structure containing details about bird.
+type birdStatus struct {
+	ready            bool
+	version          string
+	routeID          string
+	serverTime       string
+	lastBootTime     string
+	lastReconfigTime string
+}
+
+func (b *birdStatus) toNodeStatusAPI() apiv3.CalicoNodeBirdStatus {
+	return apiv3.CalicoNodeBirdStatus{
```

Comment:
Might be simpler if instead of duplicating fields on the `birdStatus` struct, we just had a single field like this:

```go
type birdStatus struct {
  status apiv3.CalicoNodeBirdStatus
}
```

Then this function becomes effectively unnecessary:

```go
func (b *birdStatus) toNodeStatusAPI() apiv3.CalicoNodeBirdStatus {
  return b.status
}
```

Reply:
I was thinking populators (in this case `birdStatus`) always has it's own internal structure and may convert to different API structures in the future. 

Say we may have another v3 API which holds part of the information from birdStatus. 

Another benefit of decoupling internal structure with API is that `show()` function may show more fields than the API does. WDYT?

Reply:
Maybe, seems like we don't have a clear need for that now which is why I was suggesting it. Doesn't seem worth it to structure the code around a hypothetical that we don't know we really will ever need.